cmds/mkdir/mkdir_test.go

```diff
@@ -0,0 +1,230 @@
+		},
+	}
+
+	regularTestCases = []struct {
+		name string
+		args []string
+		exp  []string
+	}{
+		{
+			name: "Create 1 Directory",
+			args: []string{stubDirNames[0]},
+			exp:  []string{stubDirNames[0]},
+		},
+		{
+			name: "Create 2 Directories",
+			args: stubDirNames,
+			exp:  stubDirNames,
+		},
+	}
+
+	permTestCases = []MkdirPermTestCase{
+		{
+			name:     "Default Perm",
+			args:     []string{stubDirNames[0]},
+			perm:     os.FileMode(0755 | os.ModeDir),
+			dirNames: []string{stubDirNames[0]},
+		},
+		{
+			name:     "Custom Perm in Octal Form",
+			args:     []string{"-m=0777", stubDirNames[0]},
+			perm:     os.FileMode(0777 | os.ModeDir),
+			dirNames: []string{stubDirNames[0]},
+		},
+		{
+			name:     "Custom Perm not in Octal Form",
+			args:     []string{"-m=777", stubDirNames[0]},
+			perm:     os.FileMode(0777 | os.ModeDir),
+			dirNames: []string{stubDirNames[0]},
+		},
+		{
+			name:     "Custom Perm with Sticky Bit",
+			args:     []string{"-m=1777", stubDirNames[0]},
+			perm:     os.FileMode(0777 | os.ModeDir | os.ModeSticky),
+			dirNames: []string{stubDirNames[0]},
+		},
+		{
+			name:     "Custom Perm with SGID Bit",
+			args:     []string{"-m=2777", stubDirNames[0]},
+			perm:     os.FileMode(0777 | os.ModeDir | os.ModeSetgid),
+			dirNames: []string{stubDirNames[0]},
+		},
+		{
+			name:     "Custom Perm with SUID Bit",
+			args:     []string{"-m=4777", stubDirNames[0]},
+			perm:     os.FileMode(0777 | os.ModeDir | os.ModeSetuid),
+			dirNames: []string{stubDirNames[0]},
+		},
+		{
+			name:     "Custom Perm with Sticky Bit and SUID Bit",
+			args:     []string{"-m=5777", stubDirNames[0]},
+			perm:     os.FileMode(0777 | os.ModeDir | os.ModeSticky | os.ModeSetuid),
+			dirNames: []string{stubDirNames[0]},
+		},
+		{
+			name:     "Custom Perm for 2 Directories",
+			args:     []string{"-m=5777", stubDirNames[0], stubDirNames[1]},
+			perm:     os.FileMode(0777 | os.ModeDir | os.ModeSticky | os.ModeSetuid),
+			dirNames: stubDirNames,
+		},
+	}
+)
+
+func run(c *exec.Cmd) (string, string, error) {
+	var o, e bytes.Buffer
+	c.Stdout, c.Stderr = &o, &e
+	err := c.Run()
+	return o.String(), e.String(), err
+}
+
+func printError(t *testing.T, testname string, execStmt string, out interface{}, exp interface{}) {
+	t.Logf("TEST %v", testname)
+	t.Errorf("%s\ngot:%v\nwant:%v", execStmt, out, exp)
+}
+
+func findFile(dir string, filename string) (os.FileInfo, error) {
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return nil, err
+	}
+	for _, file := range files {
+		if file.Name() == filename {
+			return file, nil
+		}
+	}
+	return nil, nil
+}
+
+func removeCreatedFiles(tmpDir string) {
+	for _, dirName := range stubDirNames {
+		os.Remove(filepath.Join(tmpDir, dirName))
+	}
+}
+
+func TestMkdir(t *testing.T) {
+	// Set Up
+	tmpDir, execPath := testutil.CompileInTempDir(t)
+	defer os.RemoveAll(tmpDir)
+	syscall.Umask(umaskDefault)
+
+	// Error Tests
+	for _, test := range errorTestCases {
```

Comment:
If these are really separate tests they ought to be separate functions, so you can see in test output which test series failed.

Reply:
Fixed